### PR TITLE
[backport] fix back-quoted constructor params with identical prefixes

### PR DESCRIPTION
**Pull request by Jasper-M**
Backport of #9008

### CODE REVIEW

File: src/reflect/scala/reflect/internal/Symbols.scala [hunk 1]
```diff
@@ -2098,18 +2098,26 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
       // handling of non-public parameters seems to change the order (see scala/bug#7035.)
       //
       // Luckily, the constrParamAccessors are still sorted properly, so sort the field-accessors using them
-      // (need to undo name-mangling, including the sneaky trailing whitespace)
+      // (need to undo name-mangling, including the sneaky trailing whitespace, and match longest first)
       //
       // The slightly more principled approach of using the paramss of the
```
Comment (jarosmpost) [inline src/reflect/scala/reflect/internal/Symbols.scala:2103]: small old typo --> The slightly more principled approach of using the **params** of the

**Comment by jarosmpost** [comment]
For what I've seen, I'm happy with this fix. But unfortunately I am not a compiler expert.

File: src/reflect/scala/reflect/internal/Symbols.scala (hunk 1 above)
Comment (Jasper-M) [reply src/reflect/scala/reflect/internal/Symbols.scala:2103]: Not 100% sure, but it's probably not a typo. `paramss` is used in the compiler to refer to multiple parameter lists, so `List[List[Param]]`.

**Review by lrytz — APPROVED** [review]
Thank you!

**Pull request merged**